src/hooks/useLocalStorage.ts

```diff
@@ -50,6 +52,37 @@ export const useLocalStorage = <D>(
   return [value, setValue]
 }
 
+type Value<T> = T | null
+
+export function useReadLocalStorage<T>(key: string): Value<T> {
+  // Get from local storage then
+  // parse stored json or return initialValue
+  const readValue = useCallback((): Value<T> => {
+    return getStorageValue(key, null)
+  }, [key])
+
+  const [storedValue, setStoredValue] = useState<Value<T>>(readValue)
+
+  // Listen if localStorage changes
+  useEffect(() => {
+    setStoredValue(readValue())
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const handleStorageChange = useCallback(
+    (event: StorageEvent | CustomEvent) => {
+      if ((event as StorageEvent)?.key && (event as StorageEvent).key !== key) {
+        return
+      }
+      setStoredValue(readValue())
+    },
+    [key, readValue],
+  )
+
+  useEvent('storage', handleStorageChange)
```

Comment:
if you're using the `useEvent` hook here, it's probably worth propagating it to `useLocalStorage` as well